### ISSUE TITLE
Should not use private IP addresses with public cloud providers

### ISSUE THREAD

**Issue by riccardomurri** [gc3-uzh-ch/elasticluster#191]
Private IP addresses are likely routed to machines on the local network, not cloud-hosted VMs.  See PR #190 for a patch against the GCE provider.  Similar patches should be applied to the other supported public clouds.

**Comment by jsharpe** [comment]
The best solution to this is actually to use ssh ProxyCommand to tunnel connections to computer nodes via the frontend node. Having compute nodes exposed to the internet with ssh access is an extra security risk.

**Comment by riccardomurri** [comment]
Good point, but I think that supporting this would be a separate enhancement: 

- create a configuration option to specify whether a public IP is *needed* (frontend) / *allowed* / *disallowed* (compute nodes)
- have the Ansible playbooks react on this and configure forwarding+masquerading on the frontend node(s).
